compressor/seata-compressor-gzip/src/main/java/io/seata/compressor/gzip/GzipUtil.java

```diff
@@ -28,48 +26,42 @@
  */
 public class GzipUtil {
 
+    private GzipUtil() {
+
+    }
+
     private static final int BUFFER_SIZE = 8192;
 
     public static byte[] compress(byte[] bytes) {
         if (bytes == null) {
             throw new NullPointerException("bytes is null");
         }
-        ByteArrayOutputStream out = null;
-        GZIPOutputStream gzip = null;
-        try {
-            out = new ByteArrayOutputStream();
-            gzip = new GZIPOutputStream(out);
+        ByteArrayOutputStream out = new ByteArrayOutputStream();
+        try (GZIPOutputStream gzip = new GZIPOutputStream(out)) {
             gzip.write(bytes);
-            gzip.close();
+            gzip.flush();
+            gzip.finish();
```

Comment:
Is it necessary here? close will call the finish method.

Reply:
> Is it necessary here? close will call the finish method.

Of course not. If you do not call the flush method and the finish method before the close method, you will not be able to write data into the stream. Since the return statement will return before the finally block, the data was not written into the stream at that time, which caused compression failure. Further decompression failed, and I wrote a new unit test method for compression and decompression testing.